src/main/java/core/basesyntax/PurchaseOperation.java

```diff
@@ -0,0 +1,15 @@
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
```

Comment:
It is recommended to use the Strategy pattern without directly passing the storage map into operations. Instead, consider creating a context that holds the storage and pass it to the strategies.